syntax = "proto3";

package user_proto;

import "google/protobuf/empty.proto";

service UserController {
  rpc List(UserListRequest) returns (stream User) {}
  rpc Create(User) returns (User) {}
  rpc Retrieve(UserRetrieveRequest) returns (User) {}
  rpc Update(User) returns (User) {}
  rpc Delete(User) returns (google.protobuf.Empty) {}
  rpc Login(UserLogin) returns (User) {}
  rpc IsUserExists(UserExists) returns (User) {}
}

message User {
  int64 id = 1;
  string username = 2;
  string first_name = 3;
  string last_name = 4;
  string email = 5;
  string password = 6;
  bool is_authenticated = 7;
  bool is_active = 8 ;
}

message UserListRequest {
}

message UserRetrieveRequest {
  int64 id = 1;
}

message UserExists {
  int64 id = 1;
}

message UserLogin {
  string username = 1;
  string password = 2;
}